BrainPortal/app/views/tasks/show.html.erb

```diff
@@ -57,6 +57,26 @@
       <% confirm += "\n\nIMPORTANT NOTE: This will also remove its archived work directory." if @task.workdir_archive %>
       <%= link_to 'Remove Task',  { :action => 'operation', :operation => 'delete', 'tasklist[]' => @task.id }, :confirm => confirm, :class  => "button menu_button", :method => :post %>
     <% end %>
+    <% if @task.status =~ /Failed*|Completed/ %>
+      <%= link_to "Remove Work Directory", url_for(:action  => :operation, :operation => 'zap_wd', 'tasklist[]' => @task.id),
+                                                  :method  => :post,
+                                                  :datatype => :script,
+                                                  :class => "button",
+                                                  :confirm =>
+                  "Are you sure you want to remove the task's work directory?"
+      %>
+    <% end %>
+
+    <!-- ARCHIVING -->
+    <% if @task.status =~ /Completed/ && @task.full_cluster_workdir.blank? == false && @task.archived_status.nil? %>
+      <%= link_to "Archive on Cluster", url_for(:action  => :operation, :operation => 'archive', 'tasklist[]' => @task.id), :method  => :post, :datatype => :script, :class => "button" %>
+      <% if @task.results_data_provider_id %>
```

Comment:
The `results_data_provider_id` has nothing to do with task archiving on file; the task's archive can be on a completely different data provider than the tasks' results. It is wrong to impose archiving to happen on the same data provider as the result, and in fact it can corrupt a lot of things (especially for DPs with dedicated data types)